src/BDCS/Export/FSTree.hs

```diff
@@ -203,29 +201,32 @@ addFileToTree root object = do
                 (Directory _, _)           -> throwError $ "Unable to to add " ++ T.unpack (filesPath object) ++
                                                            ", file already added at this location"
 
-                -- Follow the symlink, and repeat addEntryToTree with the symlink target as the new directory
-                -- to add into.
-                (Symlink s, _)             -> do
-                    -- Check and increment the link count
-                    whenM ((>= maxSymlinks) <$> get) $
-                        throwError $ "Too many levels of symbolic links while resolving " ++ T.unpack (filesPath object)
-                    modify (+1)
+                -- Allow a symlink to be added twice if it's the same symlink
+                (Symlink s1, Symlink s2)   -> if compareLinks s1 s2 then return existing
+                                              else throwError $ "Unable to add " ++ T.unpack (filesPath object) ++
```

Comment:
It might be useful to make all of these error messages say what type they are trying to add (Symlink, Directory, etc.) so that it's easier to tell what's happening without reading the code.

Reply:
Yeah I need to take a better look at these error messages. Since chunks of the tree can move around (and fail while doing so) it's not always clear what file is even causing the message. On the TODO list.